components/rag-wrapper/src/rag_wrapper/rag_ops/in_mem_graph_rag_ops.py

```
@@ -71,49 +58,31 @@ async def persist_index(self):
-                self.logger.info(f"Loaded existing index from {self.persist_dir}")
+        if not await self.index_exists():
+            # Index doesn't exist, just setup the storage context
+            storage_context = StorageContext.from_defaults(property_graph_store=self.property_graph_store)
+            rag_index = None
+            property_graph_store = None
+            self.logger.info("Index doesn't exist, storage context initialized")
+        else:
+            # Load existing index from storage
+            storage_context = StorageContext.from_defaults(persist_dir=self.persist_dir)
+            rag_index = load_index_from_storage(storage_context, embed_model=self.emb_llm if self.embed_kg_nodes else None, llm=self.completion_llm, callback_manager=self._callback_manager)
+            self.logger.info(f"Loaded existing index from {self.persist_dir}")
```

Ruff (0.15.17)

[warning] Logging statement uses f-string

(G004)